modules/mono/editor/GodotTools/GodotTools/Ides/Rider/RiderPathLocator.cs

```diff
@@ -1,185 +1,293 @@
 using System;
 using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
 using System.IO;
 using System.Linq;
-using System.Runtime.Versioning;
-using Godot;
+using System.Text.RegularExpressions;
+using JetBrains.Annotations;
 using Microsoft.Win32;
-using Newtonsoft.Json;
-using Directory = System.IO.Directory;
-using Environment = System.Environment;
-using File = System.IO.File;
-using Path = System.IO.Path;
-using OS = GodotTools.Utils.OS;
-
-// ReSharper disable UnassignedField.Local
-// ReSharper disable InconsistentNaming
-// ReSharper disable UnassignedField.Global
-// ReSharper disable MemberHidesStaticFromOuterClass
 
 namespace GodotTools.Ides.Rider
 {
+    internal enum OperatingSystemFamily
+    {
+        /// <summary>
+        ///   <para>Returned for operating systems that do not fall into any other category.</para>
+        /// </summary>
+        Other,
+
+        /// <summary>
+        ///   <para>macOS operating system family.</para>
+        /// </summary>
+        MacOSX,
+
+        /// <summary>
+        ///   <para>Windows operating system family.</para>
+        /// </summary>
+        Windows,
+
+        /// <summary>
+        ///   <para>Linux operating system family.</para>
+        /// </summary>
+        Linux
+    }
+
+    internal interface IRiderLocatorEnvironment
+    {
+        OperatingSystemFamily operatingSystemFamily { get; }
+
+        T FromJson<T>(string json);
+        void Warn(string message, Exception e = null);
+        void LogException(Exception exception);
+    }
+
     /// <summary>
     /// This code is a modified version of the JetBrains resharper-unity plugin listed under Apache License 2.0 license:
     /// https://github.com/JetBrains/resharper-unity/blob/master/unity/JetBrains.Rider.Unity.Editor/EditorPlugin/RiderPathLocator.cs
```

Comment:
This link could be updated.

```suggestion
    /// https://github.com/JetBrains/resharper-unity/blob/master/unity/EditorPlugin/RiderPathLocator.cs
```